### DISCUSSION TITLE
alerta alert and dotted field names

### DISCUSSION THREAD

**Discussion by GraFindlay** [jertel/elastalert2#752] [Q&A] [answered]
Hi, 
When trying to use a dotted field name for one of the alerta alert fields (e.g. alerta_resource) the following error occurs:

elastalert_error - {'message': "Uncaught exception running rule netops_rtcd: 'dict' object has no attribute 'source'", 'traceback': ['Traceback (most recent call last):', '  File "/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py", line 1538, in alert', '    return self.send_alert(matches, rule, alert_time=alert_time, retried=retried)', '  File "/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py", line 1633, in send_alert', '    alert.alert(matches)', '  File "/usr/local/lib/python3.10/site-packages/elastalert/alerters/alerta.py", line 51, in alert', '    alerta_payload = self.get_json_payload(matches[0])', '  File "/usr/local/lib/python3.10/site-packages/elastalert/alerters/alerta.py", line 95, in get_json_payload', "    'resource': resolve_string(self.resource, match, self.missing_text),", '  File "/usr/local/lib/python3.10/site-packages/elastalert/util.py", line 471, in resolve_string', '    string = string.format(**dd_match)', "AttributeError: 'dict' object has no attribute 'source'"], 'data': {'rule': 'netops_rtcd'}}

**conf file entry:**
alerta_resource: "{log.source.hostname}"

I'm assuming that this is python treating the field name as an object rather than a string?

Thanks,
Graham.

**Comment by jertel** [comment, marked as answer]
Use:

```
alerta_resource: "{log[source][hostname]"}
```

In future discussions please format your messages properly with backticks, so it's easier for people to help you. See #11 for more information on project guidelines and best ways to request support.

**Reply by GraFindlay** [reply]
Perfect, thanks!
